### PR TITLE
ASP-2255 Remove SBATCH param parser

### PR DESCRIPTION
**Pull request by fschuch**
#### What
Update the jobbergate-agent to use "execution parameters" for Job Submission

#### Why
Now that `execution_parameters` have been added as a field in `job_submissions`, the agent should be updated to use these.

`Task`: https://jira.scania.com/browse/ASP-2255

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: tests/jobbergate/test_submit.py [hunk 1]
```diff
@@ -415,3 +415,53 @@ def _submit_side_effect(request):
...
+        assert actual_value == desired_value
+
+    def test_extra_arguments(self):
+        """
+        Test that SlurmJobParams can be constructed with extra keyword arguments.
+        """
+        desired_value = SlurmJobParams(foo="bar", name="test-test")
+
+        actual_value = get_job_parameters(
+            slurm_parameters=dict(foo="bar"), name="test-test"
```
Comment (dusktreader) [inline tests/jobbergate/test_submit.py:464]: I would like to see a test case that covers collision of slurm_parameters and kwargs.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -7,6 +7,9 @@ This file keeps track of all notable changes to the Cluster Agent.
 Unreleased
 ----------
 
+* Removed job-script parser to extract the name and values of the SBATCH parameters contained in the file.
+* Added logic to pull `execution_parameters` from job submissions and use they as job properties when jobs are submitted.
```
Comment (dusktreader) [inline CHANGELOG.rst:11]:
```suggestion
* Added logic to pull `execution_parameters` from job submissions and use them as job properties when jobs are submitted.
```

**Review by dusktreader — CHANGES_REQUESTED** [review]
Nice work here. Added two comments for you.

File: CHANGELOG.rst (hunk 2 above)
Comment (fschuch) [reply CHANGELOG.rst]: solved in [8fa4e05](https://github.com/omnivector-solutions/cluster-agent/pull/60/commits/8fa4e0580f4941a7c65cd33bf59fe91333853929)

File: tests/jobbergate/test_submit.py (hunk 1 above)
Comment (fschuch) [reply tests/jobbergate/test_submit.py:464]: Do you mean like in the test above?

https://github.com/omnivector-solutions/cluster-agent/blob/8fa4e0580f4941a7c65cd33bf59fe91333853929/tests/jobbergate/test_submit.py#L445-L455

Comment (dusktreader) [reply tests/jobbergate/test_submit.py:464]: Oh, jeez. Sorry I didn't read this thoroughly. 🤦

**Review by dusktreader — APPROVED** [review]

**Comment by fschuch** [comment]
This is just waiting for merge because it is close related to https://github.com/omnivector-solutions/jobbergate/pull/194

**Pull request merged**